4.x cdn changes for --build set to released

sunilangadi2 opened the pull request:
Signed-off-by: Sunil Angadi <sangadi@redhat.com>

# Description
To fix RHCEPHQE-702
In order to run suite on GAed version bits of 4.x by using **--build** as **released** in CLI

sunilangadi2 commented:
@sunilkumarn417 @Ameenasuhani 
with this PR code changes tried this http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-YQUCAL/
but https://github.com/red-hat-storage/cephci/blob/b500e0beef4f087a6ddec17b3c88e76a27c40de8/ceph/ceph.py#L1730
Enabling just tools rpm will ansible enable mon and osd by itself? can you please confirm this ?
Bcz test got failed for that failure
```2021-11-29 05:52:51,264 - ceph.ceph - INFO - fatal: [ceph-spr-yqucal-node1-installer]: FAILED! => changed=false 
  attempts: 3
  failures:
  - No package ceph-test available.
  - No package ceph-mon available.
  - No package ceph-osd available.```

manasagowri commented:
> @sunilkumarn417 @Ameenasuhani with this PR code changes tried this http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-YQUCAL/ but
> 
> https://github.com/red-hat-storage/cephci/blob/b500e0beef4f087a6ddec17b3c88e76a27c40de8/ceph/ceph.py#L1730
> 
> 
> Enabling just tools rpm will ansible enable mon and osd by itself? can you please confirm this ?
> Bcz test got failed for that failure
> ```
>   attempts: 3
>   failures:
>   - No package ceph-test available.
>   - No package ceph-mon available.
>   - No package ceph-osd available.```
>   
> ```

The test got failed with same error for me as well.

psathyan commented on tests/ceph_installer/test_ansible.py:
```diff
@@ -33,6 +33,26 @@ def run(ceph_cluster, **kw):
+    # Enables cdn ansible config when --build set to relesed
+    build_type = config.get("build_type", None)
+    if build_type == "released":
+        config.get("ansi_config")["use_cdn"] = True
+        config.get("ansi_config")["ceph_repository_type"] = "cdn"
+        config.get("ansi_config")["ceph_docker_image"] = "rhceph/rhceph-4-rhel8"
+        config.get("ansi_config")["ceph_docker_image_tag"] = "latest"
+        config.get("ansi_config")["ceph_docker_registry"] = "registry.redhat.io"
+
```
We need to change this to
```
config["ansi_config"]["use_cdn"] = True
config["ansi_config"]["ceph_repository_type"] = "cdn"
config["ansi_config"]["ceph_docker_image"] = "rhceph/rhceph-4-rhel8"
config["ansi_config"]["ceph_docker_image_tag"] = "latest"
config["ansi_config"]["ceph_docker_registry"] = "registry.redhat.io"
```

sunilkumarn417 replied:
IMO, `use_cdn` was enabling all right repos in existing test cases. we may just need to enable cdn in case of `--build released`.

https://github.com/red-hat-storage/cephci/blob/master/tests/ceph_installer/test_ansible.py#L47-L64

sunilkumarn417 commented:
> @sunilkumarn417 @Ameenasuhani with this PR code changes tried this http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-YQUCAL/ but
> 
> https://github.com/red-hat-storage/cephci/blob/b500e0beef4f087a6ddec17b3c88e76a27c40de8/ceph/ceph.py#L1730
> 
> 
> Enabling just tools rpm will ansible enable mon and osd by itself? can you please confirm this ?
> Bcz test got failed for that failure
> ```
>   attempts: 3
>   failures:
>   - No package ceph-test available.
>   - No package ceph-mon available.
>   - No package ceph-osd available.```
>   
> ```

please set ceph_origin to `repository`.

sunilkumarn417 commented on ceph/ceph.py:
```diff
@@ -766,6 +766,8 @@ def setup_packages(
         """
         if not build:
             build = self.rhcs_version
+        if self.ansible_config.get("use_cdn"):
```
This might not required. `use_cdn` at test module should take care instead handling in lib modules.

sunilkumarn417 replied:
need to check regression here, if it is not working.

sunilangadi2 commented:
Thanks for the review @psathyan and @sunilkumarn417 
addressed the review and test results are here with using --build as released able to deploy 4.x tier0 with CDN bits
@sunilkumarn417 regarding `ceph-test package` will get clear clarity from dev and raise new PR for it as of now it's fine.
